Bug in user service causing compilation errors

Line 18 of user.service.ts is currently as follows:
loggedIn$ = this.loggedInSource.distinctUntilChanged().asObservable();
This generates the error
ERROR in src/app/api/user.service.ts(18,58): error TS2339: Property 'asObservable' does not exist on type 'Observable<boolean>'.
This sometimes (but not always) blocks compilation of the whole app.

--- a/src/app/api/user.service.ts
+++ b/src/app/api/user.service.ts
@@ -15,7 +15,7 @@ export class UserService {
   private user_: User;
 
   private loggedInSource = new Subject<boolean>();
-  loggedIn$ = this.loggedInSource.distinctUntilChanged().asObservable();
+  loggedIn$ = this.loggedInSource.asObservable().distinctUntilChanged();
 
   set user(value) {
     this.loggedInSource.next(value !== null);

